sig-platform-dev/charter.md

```diff
@@ -0,0 +1,375 @@
+collaborate with SIG API Machinery, who will continue to own the code.
+
+[SIG API Machinery]: https://github.com/kubernetes/community/tree/master/sig-api-machinery
+[CRD]: https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/
+[apiextensions-apiserver]: https://github.com/kubernetes/apiextensions-apiserver
+[client-go]: https://github.com/kubernetes/client-go/
+
+### Inaugural Subprojects
+
+These are subprojects that we propose to start at the inception of the SIG.
+Upon approval of the SIG charter, we will begin recruiting members and call for
+proposals to staff and flesh out these subprojects through the [KEP] process.
+
+The seed TLs will approve these initial subproject KEPs to bootstrap the set of
+subproject owners, after which subproject owners will form the primary
+decision-making body.
+
+Note that some of these are existing subprojects of other SIGs,
+and will only be transferred with consent of the owning SIG.
+
+* PDK Roadmap
+  * Define and maintain the future vision for a cohesive, first-party Platform Development Kit (PDK).
+* API Samples
+  * Create and maintain high-quality sample code for using Kubernetes API extension points.
+  * Includes sample-apiserver and sample-controller from [server-sdk][am-subprojects] in SIG-AM.
+* API Generators
+  * Create and maintain code generators for building Kubernetes APIs.
+  * Includes gengo and code-generator from [idl-schema-client-pipeline][am-subprojects] in SIG-AM.
+* [apiserver-builder](https://github.com/kubernetes-incubator/apiserver-builder)
+  * An existing subproject of SIG-AM for creating standalone or aggregated API servers.
+
+[am-subprojects]: https://github.com/kubernetes/community/tree/master/sig-api-machinery#subprojects
+
+### Experimental Subprojects
+
+These are subprojects that we believe represent experiments that are within the scope
+of the SIG, and that the SIG should facilitate and collaborate on as a way of gathering
+insight on what should go into the PDK.
+Upon approval of the SIG charter, after the inaugural subprojects are staffed with voting
+subproject owners, we will propose these subprojects through the [KEP] process.
+
+Note that some of these are existing subprojects of other SIGs,
+and will only be transferred with consent of the owning SIG.
+Others are currently external to Kubernetes, and due diligence will be exercised to
+ensure a proper transfer of project management if the KEP is approved by the SIG.
+
+Also note that even if the SIG accepts these subprojects,
+that does not make them part of the PDK.
+
+* [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
+  * A framework (generators and libraries) for building extension APIs and controllers.
+* [metacontroller](https://github.com/GoogleCloudPlatform/metacontroller)
+  * A hook-based wrapper to simplify adoption of core controller code, patterns, and best practices.
+
```

Comment:
could we please add rook's operator-kit to this list. https://github.com/rook/operator-kit